# HG changeset patch
# User Daniel J Sebald <daniel.sebald@ieee.org>
# Date 1501652618 18000
#      Wed Aug 02 00:43:38 2017 -0500
# Node ID ff2c2f2f670212f4fce7152df59ed36b8b1765bb
# Parent  9401e88f63cfae389b3f80f1eb97770dc7f46e8d
Require some plot object present to show legend for gnuplot (bug #50483).

* __gnuplot_draw_axes__.m: Add a condition to legend drawing that checks
  that at least one of the plot objects associated with all the legend children
  is visible.

diff --git a/scripts/plot/util/private/__gnuplot_draw_axes__.m b/scripts/plot/util/private/__gnuplot_draw_axes__.m
--- a/scripts/plot/util/private/__gnuplot_draw_axes__.m
+++ b/scripts/plot/util/private/__gnuplot_draw_axes__.m
@@ -1557,17 +1557,19 @@ function __gnuplot_draw_axes__ (h, plot_
       sidx_minor = 0;
     endif
     fprintf (plot_stream, "set grid linestyle %d, linestyle %d;\n",
              sidx_major, sidx_minor);
   endif
 
   if (! isempty (hlgnd) && strcmp (hlgnd.visible, "on")
       && ! isempty (hlgnd.children)
-      && any (strcmp (get (hlgnd.children, "visible"), "on")))
+      && any (strcmp (get ([get(hlgnd.children, "userdata"){:}], "visible"),
+                      "on")))
+
     if (strcmp (hlgnd.box, "on"))
       box = "box";
     else
       box = "nobox";
     endif
     if (strcmp (hlgnd.orientation, "vertical"))
       horzvert = "vertical";
     else
